usrp_drivers/n200/n200_driver.cpp

```diff
@@ -24,271 +24,339 @@
+{
+  std::vector<size_t> channels(driver_packet.channels_size());
+  for (int i=0; i<driver_packet.channels_size(); i++) {
+    channels[i] = driver_packet.channels(i);
   }
   return channels;
 }
 
-std::vector<std::vector<std::complex<float>>> make_samples(const driverpacket::DriverPacket &dp) {
-  std::vector<std::vector<std::complex<float>>> samples(dp.samples_size());
-
-  for (int i = 0 ; i < dp.samples_size(); i++) {
-    auto num_samps = dp.samples(i).real_size();
+/**
+ * @brief      Makes a set of vectors of the samples for each TX channel from the driver packet.
+ *
+ * @param[in]  driver_packet    A received driver packet from radar_control.
+ *
+ * @return     A set of vectors of TX samples for each USRP channel.
+ *
+ * Values in a protobuffer have no contiguous underlying storage so values need to be
+ * parsed into a vector.
+ */
+std::vector<std::vector<std::complex<float>>> make_tx_samples(const driverpacket::DriverPacket
+                                                                &driver_packet)
+{
+  std::vector<std::vector<std::complex<float>>> samples(driver_packet.samples_size());
+  for (int i=0; i<driver_packet.samples_size(); i++) {
+    auto num_samps = driver_packet.samples(i).real_size();
     std::vector<std::complex<float>> v(num_samps);
     samples[i] = v;
 
-    for (int j = 0; j < dp.samples(i).real_size(); j++) {
-      samples[i][j] = std::complex<float>(dp.samples(i).real(j), dp.samples(i).imag(j));
+    for (int j = 0; j < driver_packet.samples(i).real_size(); j++) {
+      samples[i][j] = std::complex<float>(driver_packet.samples(i).real(j),
+                                           driver_packet.samples(i).imag(j));
+    }
+  }
+
+  for (auto &s : samples)
+  {
+    if (s.size() != samples[0].size())
+    {
+      //TODO(keith): Handle this error.   REPLY Would this be a critical fail?
     }
   }
 
   return samples;
 }
 
-void transmit(zmq::context_t* driver_c, std::shared_ptr<USRP> usrp_d,
-                std::shared_ptr<DriverOptions> driver_options) {
-  std::cout << "Enter transmit thread\n";
+
+void transmit(zmq::context_t &driver_c, USRP &usrp_d,
+                const DriverOptions &driver_options)
+{
+  DEBUG_MSG("Enter transmit thread");
 
 
-  zmq::socket_t packet_socket(*driver_c, ZMQ_SUB);
-  packet_socket.connect("inproc://threads");
-  packet_socket.setsockopt(ZMQ_SUBSCRIBE, "", 0);
-  std::cout << "TRANSMIT: Creating and connected to thread socket in control\n";
-  zmq::message_t request;
+  zmq::socket_t driver_packet_pub_socket(driver_c, ZMQ_SUB);
+  ERR_CHK_ZMQ(driver_packet_pub_socket.connect("inproc://threads"))
+  ERR_CHK_ZMQ(driver_packet_pub_socket.setsockopt(ZMQ_SUBSCRIBE, "", 0))
+  DEBUG_MSG("TRANSMIT: Creating and connected to thread socket in control");
 
-  zmq::socket_t timing_socket(*driver_c, ZMQ_PAIR);
-  timing_socket.connect("inproc://timing");
+  zmq::socket_t receive_side_timing_socket(driver_c, ZMQ_PAIR);
```

Comment:
These can't be wrapped in a try/catch since they introduce new scope. From what I've found people just do what Im doing by checking any calls that use the object. Any ideas?